cdap-data-fabric/src/main/java/co/cask/cdap/data2/metadata/store/DefaultMetadataStore.java

```diff
@@ -150,22 +150,19 @@ public void setAuditPublisher(AuditPublisher auditPublisher) {
    */
   @Override
   public void setProperties(final MetadataScope scope, final MetadataEntity metadataEntity,
-                            final Map<String, String> properties) {
+                            final Map<String, String> currentProperties) {
     final AtomicReference<MetadataRecordV2> previousRef = new AtomicReference<>();
     execute(mds -> {
-      Map<String, String> existingProperties = mds.getProperties(metadataEntity);
-      Set<String> existingTags = mds.getTags(metadataEntity);
-      previousRef.set(new MetadataRecordV2(metadataEntity, scope, existingProperties, existingTags));
-      for (Map.Entry<String, String> entry : properties.entrySet()) {
-        mds.setProperty(metadataEntity, entry.getKey(), entry.getValue());
-      }
+      MetadataDataset.MetadataChange metadataChange = mds.setProperty(metadataEntity, currentProperties);
+      previousRef.set(new MetadataRecordV2(metadataEntity, scope, metadataChange.getExisting().getProperties(),
```

Comment:
looks like there is already an execute method in this class that returns an object. Can just use that instead of this ref

Reply:
Awesome!